modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngine.java

```diff
@@ -173,13 +175,16 @@ protected CompletableFuture<Void> onReplicasUpdate(AlterZoneEventParameters para
     // TODO: And then run the remote invoke, only if needed.
     private CompletableFuture<Void> rebalanceTriggersRecovery(long recoveryRevision) {
         if (recoveryRevision > 0) {
-            List<CompletableFuture<Void>> zonesRecoveryFutures = catalogService.zones(catalogService.latestCatalogVersion())
+            // Safe to get the latest version until lwm mechanism is implemented for zone lifecycle.
+            int catalogVersion = catalogService.latestCatalogVersion();
```

Comment:
In order to be consistent with `DistributionZoneManager` and the fact that recoveryRevision is passed to the method as a parameter I'd rather add int catalogVersion parameter next to it.